src/webgpu/shader/validation/expression/unary/address_of_and_indirection.spec.ts

```diff
@@ -0,0 +1,242 @@
+export const description = `
+Validation tests for unary address-of and indirection (dereference)
+`;
+
+import { makeTestGroup } from '../../../../../common/framework/test_group.js';
+import { keysOf } from '../../../../../common/util/data_tables.js';
+import { ShaderValidationTest } from '../../shader_validation_test.js';
+
+export const g = makeTestGroup(ShaderValidationTest);
+
+const kAddressSpaces = ['function', 'private', 'workgroup', 'uniform', 'storage'];
+const kAccessModes = ['read', 'read_write'];
+const kStorageTypes = ['bool', 'u32', 'i32', 'f32', 'f16'];
+const kCompositeTypes = ['array', 'struct', 'vec', 'mat'];
+const kDerefTypes = {
+  deref_address_of_identifier: {
+    wgsl: (id: string, ptr: string) => `(*(&${id}))`,
```

Comment:
nit: doesn't actually look like you vary the arguments passed to these functions. You could just turn these functions to string literals using `a` for `id` and `p` for `ptr`.